app/src/main/java/tech/relaycorp/courier/data/model/MessageAddress.kt

```diff
@@ -33,5 +38,31 @@ sealed class MessageAddress {
     }
 }
 
-data class PublicMessageAddress(val publicValue: String) : MessageAddress()
+data class PublicMessageAddress(val publicValue: String) : MessageAddress() {
+    @Suppress("BlockingMethodInNonBlockingContext") // Needed for URL()
+    @Throws(PublicAddressResolutionException::class)
+    suspend fun resolve(doHClient: DoHClient): String {
+        val originalURL = URL(publicValue)
+        val srvName = "_rcrc._tcp.${originalURL.host}"
```

Comment:
For example, if `${originalURL.host} == "frankfurt.relaycorp.cloud"`: https://mxtoolbox.com/SuperTool.aspx?action=srv%3a_rcrc._tcp.frankfurt.relaycorp.cloud&run=toolpage